File: momen/src/export/csvExporter.ts
/**
 * Momen — CSV Exporter
 *
 * Generates a standard CSV file with one row per marker.
 * Columns: Marker Number, Timecode, Duration, Note, Sync Point
 */

import { Marker, FrameRate } from '../types';
import { SYNC_NOTE } from '../constants';

export function generateCSV(markers: Marker[], fps: FrameRate): string {
  const header = 'Marker Number,Timecode,Duration,Note,Sync Point';

  const rows = markers.map((marker) => {
    const duration = '00:00:01:00'; // Conventional 1-second SMPTE duration
    const note = marker.isSyncPoint
      ? escapeCSV(SYNC_NOTE)
      : escapeCSV(marker.note);
    const syncPoint = marker.isSyncPoint ? 'TRUE' : 'FALSE';

    return `${marker.markerNumber},${marker.timecodeSmpte},${duration},${note},${syncPoint}`;
  });

  return [header, ...rows].join('\n');
}

function escapeCSV(value: string): string {
  if (value.includes(',') || value.includes('"') || value.includes('\n')) {
    return `"${value.replace(/"/g, '""')}"`;
  }
  return value;
}

File: momen-ios/MomenKit/Sources/MomenKit/CSVExporter.swift
import Foundation

/// CSV export — one row per marker.
/// Columns: Marker Number, Timecode, Duration, Note, Sync Point
public enum CSVExporter {

    public static func generate(markers: [ExportMarker], fps: FrameRate) -> String {
        let header = "Marker Number,Timecode,Duration,Note,Sync Point"
        // 1-second duration uses the same separator convention as the row's timecode.
        let duration = fps.isDropFrame ? "00:00:01;00" : "00:00:01:00"

        let rows = markers.map { marker -> String in
            let note = marker.isSyncPoint
                ? escape(MomenConstants.syncNote)
                : escape(marker.note)
            let syncPoint = marker.isSyncPoint ? "TRUE" : "FALSE"
            return "\(marker.markerNumber),\(marker.timecodeSmpte),\(duration),\(note),\(syncPoint)"
        }

        return ([header] + rows).joined(separator: "\n")
    }

    static func escape(_ value: String) -> String {
        if value.contains(",") || value.contains("\"") || value.contains("\n") {
            return "\"\(value.replacingOccurrences(of: "\"", with: "\"\""))\""
        }
        return value
    }
}
